library/std/src/sys/alloc/vexos.rs

```diff
@@ -0,0 +1,118 @@
+// FIXME(static_mut_refs): Do not allow `static_mut_refs` lint
+#![allow(static_mut_refs)]
+
+use crate::alloc::{GlobalAlloc, Layout, System};
+use crate::ptr;
+use crate::sync::atomic::{AtomicBool, Ordering};
+
+// symbols defined in the target linkerscript
+unsafe extern "C" {
+    static mut __heap_start: u8;
+    static mut __heap_end: u8;
+}
+
+static mut DLMALLOC: dlmalloc::Dlmalloc<Vexos> = dlmalloc::Dlmalloc::new_with_allocator(Vexos);
```

Comment:
Is it possible to use `Mutex` instead of manually reimplementing one in `mod lock`?

Reply:
This module is more or less a port of the wasm dlmalloc implementation, though given that we are a single-threaded target I think we should just no-op the `lock` function here.